controller.go

```diff
@@ -15,26 +15,120 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
 	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
 )
 
 // The controller implements the web API of the cloud orchestrator. It parses
 // and validates requests from the client and passes the information to the
 // relevant modules
 type Controller struct {
+	ss SignalingServer
```

Comment:
nit: should we have more descriptive field names? Go advocates for one letter names when scope is smaller, but field names could be used anywhere.

Reply:
absolutely!